WcaOnRails/app/models/wcif.rb

```diff
@@ -13,6 +13,23 @@ def json
     {
       "formatVersion" => "1.0",
       "id" => competition.id,
+      "events" => competition.competition_events.map(&method(:competition_event_to_json)),
+    }
+  end
+
+  private def competition_event_to_json(competition_event)
```

Comment:
Interesting argument. I'd say that even if some cases require special treatment, this would serve well as the normal pattern. But that's mostly a theoretical assertion.